app/javascript/packages/document-capture/components/in-person-full-address-search.tsx

```diff
@@ -0,0 +1,227 @@
+    sundayHours: po.sunday_hours,
+    weekdayHours: po.weekday_hours,
+    isPilot: !!po.is_pilot,
+  }));
+
+const requestUspsLocations = async (address: LocationQuery): Promise<FormattedLocation[]> => {
+  const response = await request<PostOffice[]>(LOCATIONS_URL, {
+    method: 'post',
+    json: { address: transformKeys(address, snakeCase) },
+  });
+
+  return formatLocations(response);
+};
+
+function useUspsLocations() {
+  const [locationQuery, setLocationQuery] = useState<LocationQuery | null>(null);
+  const validatedAddressFieldRef = useRef<HTMLFormElement>(null);
+  const validatedCityFieldRef = useRef<HTMLFormElement>(null);
+  const validatedStateFieldRef = useRef<HTMLFormElement>(null);
+  const validatedZipCodeFieldRef = useRef<HTMLFormElement>(null);
+
+  const handleLocationSearch = useCallback(
+    (event, addressInput, cityInput, stateInput, zipCodeInput) => {
+      event.preventDefault();
+      validatedAddressFieldRef.current?.setCustomValidity('');
+      validatedAddressFieldRef.current?.reportValidity();
+      validatedCityFieldRef.current?.setCustomValidity('');
+      validatedCityFieldRef.current?.reportValidity();
+      validatedStateFieldRef.current?.setCustomValidity('');
+      validatedStateFieldRef.current?.reportValidity();
+      validatedZipCodeFieldRef.current?.setCustomValidity('');
+      validatedZipCodeFieldRef.current?.reportValidity();
+
+      if (
+        addressInput.trim().length === 0 ||
+        cityInput.trim().length === 0 ||
+        stateInput.trim().length === 0 ||
+        zipCodeInput.trim().length === 0
+      ) {
+        return;
+      }
+
+      setLocationQuery({
+        address: `${addressInput}, ${cityInput}, ${stateInput} ${zipCodeInput}`,
+        streetAddress: addressInput,
+        city: cityInput,
+        state: stateInput,
+        zipCode: zipCodeInput,
+      });
+    },
+    [],
+  );
+
+  const {
+    data: locationResults,
+    isLoading: isLoadingLocations,
+    error: uspsError,
+  } = useSWR([locationQuery], ([address]) => (address ? requestUspsLocations(address) : null));
+
+  return {
+    locationQuery,
+    locationResults,
+    uspsError,
+    isLoading: isLoadingLocations,
+    handleLocationSearch,
+    validatedAddressFieldRef,
+    validatedCityFieldRef,
+    validatedStateFieldRef,
+    validatedZipCodeFieldRef,
+  };
+}
+
+interface FullAddressSearchProps {
+  registerField?: RegisterFieldCallback;
+  onFoundLocations?: (
+    address: LocationQuery | null,
+    locations: FormattedLocation[] | null | undefined,
+  ) => void;
+  onLoadingLocations?: (isLoading: boolean) => void;
+  onError?: (error: Error | null) => void;
+  disabled?: boolean;
+}
+
+function FullAddressSearch({
+  registerField = () => undefined,
+  onFoundLocations = () => undefined,
+  onLoadingLocations = () => undefined,
+  onError = () => undefined,
+  disabled = false,
+}: FullAddressSearchProps) {
+  // todo: should we get rid of verbose 'input' word?
```

Comment:
Is this still TODO? Can we make a decision? No strong preference from me, I could be fine with either dropping the suffix, or maybe even "value" as a suffix instead, since "input" to me is the element itself.

Reply:
fixed in 68152ca3f49551e20d957bb34a7452b45f0fd2ef